Generate.py

```diff
@@ -378,6 +378,46 @@ def roll_linked_options(weights: dict) -> dict:
     return weights
 
 
+def handle_trigger_math(math_options: Union[dict, string], option_name: str, weights: dict):
+    options = copy.deepcopy(math_options)
+    if not isinstance(options, dict):
+        options = {str(options): 1}
+    new_dict = {}
+    for name, value in options.items():
+        value_list = []
+        if not isinstance(name, str):
+            raise Exception(f'Mathematical trigger malformed: {name} is not a string.')
```

Comment:
Following the style guide, these should use doublequotes
```suggestion
            raise Exception(f"Mathematical trigger malformed: {name} is not a string.")
```